Kitodo-SRU-Import/src/main/java/org/kitodo/sruimport/ResponseHandler.java

```diff
@@ -0,0 +1,144 @@
+import org.jdom2.output.XMLOutputter;
+import org.kitodo.api.externaldatamanagement.Record;
+import org.kitodo.api.externaldatamanagement.SearchResult;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+class ResponseHandler {
+
+    private static final Logger logger = LogManager.getLogger(ResponseHandler.class);
+
+    private static DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+    private static XMLOutputter xmlOutputter = new XMLOutputter();
+
+    static {
+        documentBuilderFactory.setNamespaceAware(true);
+        xmlOutputter.setFormat(Format.getPrettyFormat());
+    }
+
+    private static final String MODS_NAMESPACE = "http://www.loc.gov/mods/v3";
+    private static final String SRW_NAMESPACE = "http://www.loc.gov/zing/srw/";
+    private static final String SRW_RECORD_TAG = "record";
+    private static final String SRW_NUMBER_OF_RECORDS_TAG = "numberOfRecords";
+    private static final String MODS_TAG = "mods";
+    private static final String RECORD_ID_TAG = "recordIdentifier";
+    private static final String RECORD_TITLE_TAG = "title";
+
+    /**
+     * Private constructor.
+     */
+    private ResponseHandler() {
+    }
+
+    /**
+     * Create and return SearchResult for given HttpResponse.
+     * @param response HttpResponse for which a SearchResult is created
+     * @return SearchResult created from given HttpResponse
+     */
+    static SearchResult getSearchResult(HttpResponse response) {
+        SearchResult result = new SearchResult();
+        Document resultDocument = transformResponseToDocument(response);
+        if (Objects.nonNull(resultDocument)) {
+            result.setHits(extractHits(resultDocument));
+            result.setNumberOfRecords(extractNumberOfRecords(resultDocument));
+        }
+        return result;
+    }
+
+    /**
+     * Transform given HttpResponse into Document and return it.
+     * @param response HttpResponse that is transformed into a Document
+     * @return Document into which given HttpResponse has been transformed
+     */
+    public static Document transformResponseToDocument(HttpResponse response) {
+        HttpEntity entity = response.getEntity();
+        if (Objects.nonNull(entity)) {
+            try {
+                return parseXML(entity.getContent());
+            } catch (IOException e) {
+                logger.error(e);
+            }
+        }
+        return null;
+    }
+
+    private static Document parseXML(InputStream xmlSteam) {
+        Document document = null;
+
+        try {
+            DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+            document = documentBuilder.parse(new InputSource(xmlSteam));
+        } catch (ParserConfigurationException | IOException | SAXException e) {
+            logger.error(e);
+        }
+
+        return document;
```

Comment:
Null returns should be avoided. Is it possible to return here for e.g. empty Document? 

In case of exeception document is null.

Reply:
Done.